ui/src/main/webapp/app/logo/logo-service.js

```diff
@@ -38,7 +38,7 @@ angular.module('app').factory('logo', ['properties', '$q', 'config', function(pr
                 if (!props['gaffer.properties.app.logo.src']) {
                     defer.resolve(null);
                 } else {
-                    var srcPrefix = conf.restEndpoint.replace(/\/$/, ''); // remove trailing slash
+                    var srcPrefix = common.parseUrl(conf.restEndpoint).replace(/\/$/, ''); // remove trailing slash
```

Comment:
The parseUrl method just adds http onto the url. Perhaps it should be a bit cleverer than that. It could do something like:
```js
window.location.protocol
```
to get the protocol of the current location and use that protocol instead of assuming it will be http.